lib/MusicBrainz/Server/Data/Collection.pm

```diff
@@ -136,33 +136,51 @@ sub get_first_collection {
     return $self->sql->select_single_value($query, $editor_id);
 }
 
-sub find_all_by_editor {
-    my ($self, $id, $show_private, $entity_type) = @_;
-    my $extra_conditions = (defined $entity_type) ? "AND ct.entity_type = '$entity_type'" : "";
-    if (!$show_private) {
-        $extra_conditions .= "AND editor_collection.public=true ";
-    }
+sub find_by {
+    my ($self, $opts, $limit, $offset) = @_;
 
-    my $query = "SELECT " . $self->_columns . "
-                 FROM " . $self->_table . "
-                    JOIN editor_collection_type ct
-                        ON editor_collection.type = ct.id
-                 WHERE editor=? $extra_conditions";
+    my (@joins, @conditions, @args);
 
-    $query .= "ORDER BY musicbrainz_collate(editor_collection.name)";
-    $self->query_to_list($query, [$id]);
-}
+    if (my $editor_id = $opts->{editor_id}) {
+        push @conditions, 'editor = ?';
+        push @args, $editor_id;
+    }
 
-sub find_all_by_entity {
-    my ($self, $type, $id) = @_;
-    my $query = "SELECT " . $self->_columns . "
-                 FROM " . $self->_table . "
-                    JOIN editor_collection_$type ce
-                        ON editor_collection.id = ce.collection
-                 WHERE ce.$type = ? ";
+    if (my $entity_type = $opts->{entity_type}) {
+        push @joins, 'JOIN editor_collection_type ct ON editor_collection.type = ct.id';
+        push @conditions, 'ct.entity_type = ?';
```

Comment:
Sorry, thought you just meant here and didn't check the other one. Fixed in https://github.com/metabrainz/musicbrainz-server/commit/cacaf237e110045acbd3106e8530b25fc0702359